tests/test_spark.py

```diff
@@ -66,11 +71,53 @@ def test_jar():
 @pytest.mark.sanity
 def test_teragen():
     if _hdfs_enabled():
-        jar_url = "https://downloads.mesosphere.io/spark/examples/spark-terasort-1.0-jar-with-dependencies_2.11.jar"
-        _run_tests(jar_url,
-                   "1g hdfs:///terasort_in",
-                   "Number of records written",
-                   ["--class", "com.github.ehiggs.spark.terasort.TeraGen"])
+        _run_teragen()
+
+
+def _run_teragen():
```

Comment:
I think all the other private functions are at the bottom of this file.  Can we maintain that organization?

Actually, I think ideally, the terasort test would be in a separate `test_soak.py` file.  Then the tests would be more organized by their application.